Biology.java

```diff
@@ -0,0 +1,346 @@
+                    currentSite.useDelimiter(",");
+
+                    currentSite.next(); // ignore site number
+
+                    for (int j = 0; j < numberOfAminoAcids; j++) {
+                        double probability = currentSite.nextDouble();
+                        aminoAcidPreference[i][j] = probability;
+                    }
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        public double[][] getAminoAcidPreference() {
+            return aminoAcidPreference;
+        }
+    }
+
+    /**
+     * Possible mutations that can occur at a single nucleotide site
+     */
+    public enum MutationType {
+        MUTATION();
+        public final Map<Character, double[]> transitionTranversionProbability;
+
+        MutationType() {
+            this.transitionTranversionProbability = new HashMap<>()  {{
+                double transition = 0.5 / (Parameters.transitionTransversionRatio + 1.0);
+                double transversion = Parameters.transitionTransversionRatio / (Parameters.transitionTransversionRatio + 1.0);
+                // key: nucleotide
+                // value: int[] that gives the probability of being < to the index which corresponds to [A, C, G, T]
+
+                // Example:
+                // Parameters.transitionTransversionRatio: 5.0
+                // A: {0, 0.5/6, 5.5/6.0, 1.0}
+                // C: {0.5/6.0, 0.5/6.0, 1.0/6.0, 1.0}
+                // G: {5.0/6.0, 5.5/6.0, 5.5/6.0, 1.0}
+                // T: {0.5/6.0, 5.5/6.0, 1.0, 1.0}
+
+                put('A', new double[]{0, transition, transition + transversion, 1.0});
+                put('C', new double[]{transition, transition, transition + transition, 1.0});
+                put('G', new double[]{transversion, transversion + transition, transversion + transition, 1.0});
+                put('T', new double[]{transition, transition + transversion, 1.0, 1.0});
+            }};
+        }
+
+        /**
+         * Returns a mutant nucleotide using the transition/transversion ratio.
+         *
+         * @param originalNucleotideToMutate the original nucleotide to mutate
+         * @return the new nucleotide to change the original nucleotide to
+         */
+        public char getNucleotide(char originalNucleotideToMutate) {
+            double[] transitionTransversion = this.transitionTranversionProbability.get(originalNucleotideToMutate);
+
+            // choose a random number between 0-1
+            double randomNum = Math.random();
+            int indexAlphabet = 0;
+
+            for (int i = 0; i < 4; i++) {
+                if (randomNum < transitionTransversion[i]) {
+                    indexAlphabet = i;
+                    break;
+                }
+            }
+
+            return Parameters.AlphabetType.NUCLEOTIDES.getValidCharacters().charAt(indexAlphabet);
+        }
+    }
+
+    /**
+     * DNA and RNA codon table
```

Comment:
Let's just call this a "DNA codon table" since an RNA codon table would have "U" instead of "T"